SilMock/Google/Service/Directory/Resource/GroupsAliases.php

```diff
@@ -0,0 +1,90 @@
+    }
+
+    public function delete(string $groupKey, string $alias, array $optParams = [])
+    {
+        $groupRecords = $this->getRecords();
+        foreach ($groupRecords as $groupRecord) {
+            $groupRecordData = json_decode($groupRecord['data'], true);
+            if (
+                $groupRecordData['primaryEmail'] === $groupKey
+                && $groupRecordData['alias'] === $alias
+            ) {
+                $this->deleteRecordById($groupRecord['id']);
+            }
+        }
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function insert(
+        string $groupKey,
+        GoogleDirectory_GroupsAlias $postBody,
+        array $optParams = []
+    ): GoogleDirectory_GroupsAlias {
+        if ($this->isNewGroupAlias($groupKey, $postBody->getAlias())) {
+            $postBody['id'] = $postBody['id'] ?? microtime();
+            $dataAsJson = json_encode(get_object_vars($postBody));
+            $this->addRecord($dataAsJson);
+        }
+
+        $newGroupAlias = new GoogleDirectory_GroupsAlias();
+        ObjectUtils::initialize($newGroupAlias, $postBody);
+
+        return $newGroupAlias;
+    }
+
+    /**
+     * @param $optParams -- Initial implementation ignores this.
+     * @return GoogleDirectory_GroupsAliases
+     */
+    public function listGroupsAliases(string $groupKey, array $optParams = []): GoogleDirectory_GroupsAliases
+    {
+        $groupAliases = [];
+        $directoryGroupAliasRecords = $this->getRecords();
+        $groupAliasCounter = 0;
+        foreach ($directoryGroupAliasRecords as $groupAliasRecord) {
+            $groupRecordData = json_decode($groupAliasRecord['data'], true);
+            if ($groupRecordData['primaryEmail'] === $groupKey) {
+                $currentGroup = new GoogleDirectory_GroupsAlias();
+                ObjectUtils::initialize($currentGroup, $groupRecordData);
+                $groupAliases[] = $currentGroup;
+                $groupAliasCounter = $groupAliasCounter + 1;
+            }
+        }
+        $groupsAliasesObject = new GoogleDirectory_GroupsAliases();
+        $groupsAliasesObject->setEtag('');
+        $groupsAliasesObject->setKind('groupAliases');
+        $groupsAliasesObject->setAliases($groupAliases);
+        return $groupsAliasesObject;
+    }
+
+    protected function isNewGroupAlias(string $groupKey, string $alias): bool
+    {
+        $mockGroupsObject = new GroupsAliases($this->dbFile);
+        $groupsAliasesObject = $mockGroupsObject->listGroupsAliases($groupKey);
+        /** @var GoogleDirectory_GroupsAlias[] $aliasObjects */
+        $aliasObjects = $groupsAliasesObject->getAliases();
+        $groupAliasesEmailAddresses = [];
+        foreach ($aliasObjects as $aliasObject) {
+            $groupAliasesEmailAddresses[] = mb_strtolower($aliasObject->getAlias());
+        }
+        $lowercaseAliasEmailAddress = mb_strtolower($alias);
+        return ! in_array($lowercaseAliasEmailAddress, $groupAliasesEmailAddresses);
```

Comment:
Probably not too significant performance-wise for our use cases, but rather than putting everything in an array and checking `in_array`, you could just check in the foreach loop. Such as:

```suggestion
        $found = false;
        foreach ($aliasObjects as $aliasObject) {
            if (mb_strtolower($aliasObject->getAlias()) === mb_strtolower($alias)) {
                $found = true;
                break;
            }
        }
        return $found;
```

Though I believe `==` might be more comparable with `in_array`.